src/decoder.js

```diff
@@ -31,11 +39,13 @@ export const decodeInputData = (inputData) => {
  *
  */
 
-export const decodeLogData = (logData) => {
+export const decodeLogData = (logData, address) => {
   if (!consensysDecode.getABIs().length) throw new Error('No ABIs added to system');
-  if (logData !== undefined) {
-    const decodedlogs = consensysDecode.decodeLogs(logData);
-    if (decodedlogs === undefined) { throw new Error('Problem with transaction Decodedlogs'); } else return decodedlogs;
+  if (!logData || !isAddress(address)) throw new Error('logData or address are undefined');
+  const decodedlogs = consensysDecode.decodeLogs(logData);
+  if (!decodedlogs[0]) {
```

Comment:
this will raise Error 'Array is undefined" in case there's no results;
use this `!decodedlogs.length` instead.

you can also do something like this.
return decodedlogs.length?decodedlogs:logData;